cpmpy/expressions/globalconstraints.py

```diff
@@ -197,6 +198,21 @@ def value(self):
         return len(set(vals)) == len(vals)
 
 
+class AllDifferentLists(GlobalConstraint):
+    def __init__(self, lists):
+        super().__init__("alldifferent_lists", [flatlist(lst) for lst in lists])
+
+    def decompose(self):
+        """Returns the decomposition
+        """
+        from .python_builtins import any
+        return [any([var1 != var2 for (var1, var2) in zip(vars1, vars2)]) for vars1, vars2 in all_pairs(self.args)], []
```

Comment:
Rename vars1, vars2 to lst1, lst2?
Also, I think rewriting this to a for loop might be easier to read
```python
cons = []
for lst1, lst2 in all_pairs(self.args):
     cons.append(cpm_any(var1 != var2) for var1,var2 in zip(lst1, lst2))
```